src/AES.cpp

```diff
@@ -50,8 +50,8 @@ unsigned char *AES::DecryptECB(unsigned char in[], unsigned int inLen,
   return out;
 }
 
-unsigned char *AES::EncryptCBC(unsigned char in[], unsigned int inLen,
-                               unsigned char key[], unsigned char *iv) {
+unsigned char *AES::EncryptCBC(const unsigned char in[], unsigned int inLen,
```

Comment:
Why not
`const unsigned int inLen`  
`const unsigned char *iv`

Reply:
True, the `iv` argument can also be `const`. Sorry, I missed it 😉 

I've pushed another commit to this PR that makes the `iv` arguments `const` too.

About the `unsigned int inLen` argument: the `const` modifier wouldn't have a real advantage for native types that are passed by value. The called function cannot change the value of the argument, so declaring it `const` would just have an effect inside the function. Which I think wouldn't be of any help. The advantage of using `const` on arguments is that you can judge from the outside which passed pointers or references can be changed by a function.

Reply:
Ah, right, I made a mistake with `void AES::CheckLength(const unsigned int len)` 🙈 Good catch! And sorry for the mistake.

I removed `const` from that.